extensions/src/AWSSDK.Extensions.Configuration.AWSSystemsManager/AWSSystemsManagerConfigurationProvider.cs

```diff
@@ -0,0 +1,119 @@
+
+// ReSharper disable once CheckNamespace
+namespace Amazon.Extensions.Configuration.AWSSystemsManager
+{
+    /// <inheritdoc />
+    /// <summary>
+    /// An AWS Systems Manager Parameter Store based <see cref="IConfigurationProvider" />.
+    /// </summary>
+    public class AWSSystemsManagerConfigurationProvider : ConfigurationProvider
+    {
+        private readonly AWSSystemsManagerConfigurationSource _configurationSource;
+        private readonly IAWSSystemsManagerProcessor _awsSystemsManagerProcessor;
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Initializes a new instance with the specified source.
+        /// </summary>
+        /// <param name="configurationSource">The <see cref="IConfigurationSource"/> used to retrieve values from AWS Systems Manager Parameter Store</param>
+        public AWSSystemsManagerConfigurationProvider(AWSSystemsManagerConfigurationSource configurationSource) : this(configurationSource, new AWSSystemsManagerProcessor())
+        {
+        }
+
+        public AWSSystemsManagerConfigurationProvider(AWSSystemsManagerConfigurationSource configurationSource, IAWSSystemsManagerProcessor awsSystemsManagerProcessor)
+        {
+            _configurationSource = configurationSource ?? throw new ArgumentNullException(nameof(configurationSource));
+            _awsSystemsManagerProcessor = awsSystemsManagerProcessor;
+            if (configurationSource.AwsOptions == null) throw new ArgumentNullException(nameof(configurationSource.AwsOptions));
+            if (configurationSource.Path == null) throw new ArgumentNullException(nameof(configurationSource.Path));
+
+            if (configurationSource.ReloadAfter != null)
+            {
+                ChangeToken.OnChange(() =>
+                {
+                    var cancellationTokenSource = new CancellationTokenSource(configurationSource.ReloadAfter.Value);
+                    var cancellationChangeToken = new CancellationChangeToken(cancellationTokenSource.Token);
+                    return cancellationChangeToken;
+                }, async () => await LoadAsync(true));
+            }
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Loads the AWS Systems Manager Parameters.
+        /// </summary>
+        public override void Load() => LoadAsync(false).ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync(bool reload)
+        {
+            try
+            {
+                var path = _configurationSource.Path;
+                var awsOptions = _configurationSource.AwsOptions;
+                var parameters = await _awsSystemsManagerProcessor.GetParametersByPathAsync(awsOptions, path);
+
+                Data = ProcessParameters(parameters, path);
+
+                OnReload();
+            }
+            catch (Exception ex)
+            {
+                var optional = this._configurationSource.Optional;
+                if (optional | reload) return;
```

Comment:
I feel like I did this for a reason as I never use a single | in normal use, but tbh I don't recall why. 
I'll take a look later and if it can be changed I will.